pkgs/applications/misc/kiwix/default.nix

```diff
@@ -31,6 +29,9 @@ let
     sha256 = "0sqk0vdwjq44jxbbkj1cy8qykrmafs1sickzldb2w2nshsnjshhg";
   };
 
+  xapian = callPackage ../../../development/libraries/xapian { stdenv = overrideCC stdenv gcc49; };
```

Comment:
I think you could also pass along the input `stdenv`, the cc has already been overridden.